### ISSUE TITLE
Fedora 34 does not wait for network for Clevis/Tang LUKS encryption

### ISSUE THREAD

**Issue by rugk** [coreos/fedora-coreos-tracker#827]
**Describe the bug**

**Reproduction steps**
The usual stuff from what I try the whole time:
https://discussion.fedoraproject.org/t/no-network-ipv4-connectivity-via-ethernet-at-first-boot-for-luks-clevis-setup/29260/9
TL;DR: Encrypt your root FS with LUKS and Clevis/Tang.

And it’s draining it does not work… yet again…

See https://github.com/coreos/fedora-coreos-tracker/issues/692

**Expected behavior**

**Actual behavior**
A clear and concise description of what actually happened.

**System details**
 - Bare Metal
 -  tested with current testing and stable branch
 - next seems to be currently: ` fedora-coreos-34.20210503.1.0-metal.x86_64.raw.xz` 

**Ignition config**
still the same [as initially said](https://discussion.fedoraproject.org/t/no-network-ipv4-connectivity-via-ethernet-at-first-boot-for-luks-clevis-setup/29260):
```yml
variant: fcos
version: 1.3.0
# ...
  luks:
    # encrypted root
    - name: root
      label: luks-root
      device: /dev/disk/by-partlabel/root
      clevis:
        tpm2: false
        threshold: 1
        tang:
          - url: # …
            thumbprint: # …
      wipe_volume: true
  # resize root volume to 100% of disk
  # https://docs.fedoraproject.org/en-US/fedora-coreos/storage/#_sizing_the_root_partition
storage:
  disks:
    - device: /dev/nvme0n1
      partitions:
        - label: root
          number: 4
          # 0 means to use all available space
          size_mib: 0
          resize: true
  filesystems:
    - device: /dev/mapper/root
      format: btrfs
      wipe_filesystem: true
      label: root
```

Ignition v2.9

**Additional information**

Here more system logs: https://gist.github.com/rugk/039ac5c0fb6ad7590ff88de82277c898
I have more logs, so if you need them, I' could share them privately.

Maybe it’s the same as [my post before](https://discussion.fedoraproject.org/t/no-network-ipv4-connectivity-via-ethernet-at-first-boot-for-luks-clevis-setup/29260/9?u=rugk)? For logs of that, see the gist: https://gist.github.com/rugk/44b4f02f3133130a8f0973ae46cd79b8 (previously reported as https://github.com/coreos/fedora-coreos-tracker/issues/814)

**Comment by jlebon** [comment]
>  Or is there a workaround I can use for now to not try provisioning 100 times and play the lottery whether I am lucky enough not to meet that race condition of having a network that appears too slow for Clevis to bind the encryption?

Yes, try having your Ignition config merge a remote Ignition config. E.g.

```json
{
  "ignition": {
    "version": "3.1.0",
    "config": {
      "merge": [{
        "source": "https://example.com/foobar.ign"
      }]
    }
  }
}
```

That'll force the fetch stage to wait for networking before moving to the disks stage.

**Comment by rugk** [comment]
Could it be the same issue [affects an installer Ignition file as documented here](https://coreos.github.io/coreos-installer/customizing-install/#custom-coreos-installer-invocation)?

```
variant: fcos
version: 1.3.0

# source: https://coreos.github.io/coreos-installer/customizing-install/#custom-coreos-installer-invocation
# starts installation directly after booting to /dev/nvme0n1
# cf: https://github.com/coreos/fedora-coreos-docs/issues/245#issuecomment-775546857
systemd:
  units:
    - name: install.service
      enabled: true
      contents: |
        [Unit]
        Description=Run CoreOS Installer
        Requires=coreos-installer-pre.target
        After=coreos-installer-pre.target
        OnFailure=emergency.target
        OnFailureJobMode=replace-irreversibly

        # Can be removed if install doesn't reference remote resources with
        # --stream, --image-url, or --ignition-url
        After=network-online.target
        Wants=network-online.target

        [Service]
        Type=oneshot
        ExecStart=/usr/bin/coreos-installer install /dev/nvme0n1 --insecure-ignition --ignition-url http://[…] --delete-karg 'console=ttyS0,115200n8'
        ExecStart=/usr/bin/systemctl --no-block reboot
        StandardOutput=kmsg+console
        StandardError=kmsg+console

        [Install]
        RequiredBy=default.target
 ```
 
 I also get no network and just an emergency shell and no network IP. Again the next stream here: `fedora-coreos-34.20210503.1.0`.

**Comment by dustymabe** [comment]
Am I clear that we are just waiting on `ignition-2.10.1` here? Should be in the `testing` and `next` stream releases next week.

**Comment by rugk** [comment]
Yes I guess. And thanks, that's great to hear…

> Workaround: Adding a hackish ExecStart=sleep 30s for the start “fixed” the issue. laughing

BTW, for whatever reason this now stopped working. Even if I increased the wait period to 1min, it fails.
I can see the login screen of CoreOS and it has generated some ssh keys, but it still does not have any network connection???

**Comment by rugk** [comment]
Likely having the same network error [when trying to layer a package as a service, it could not fetch the mirror URLs](https://discussion.fedoraproject.org/t/layering-packages-via-rpm-ostree-as-part-of-provisioning/29862/10?u=rugk).

**Comment by dustymabe** [comment]
The fix for this went into stable stream release `34.20210518.3.0`.

**Comment by rugk** [comment]
Okay great. I hope this also fixes the network errors for normal boots after the provisioning. :smiley: (I had errors where I had to boot from an USB drive and then reboot to make it detect a network properly.)

**Comment by bgilbert** [comment]
@rugk If you're still having network errors, please open a new issue with details.

**Comment by rugk** [comment]
I've tested it today and I can confirm it works. So thanks a lot! :tada:

**Comment by rugk** [comment]
Unfortunately, the error is here again: https://github.com/coreos/fedora-coreos-tracker/issues/852

**Comment by ShamimIslam** [comment]
> I've tested it today and I can confirm it works. So thanks a lot! tada

Hello. I realize this issue is closed, but I'm having similar problems getting clevis to work on Fedora 34. I'm sure my setup is not the same as yours. I'm willing to restart on a vanilla installation. My Fedora 34 is barebones from the live-install. Can you please suggest the steps to follow to get my setup to work with clevis NBDE?

**Comment by dustymabe** [comment]
@ShamimIslam - please open a new issue with all relevant information.

**Comment by bgilbert** [comment]
Also, note that this is the bug tracker for Fedora CoreOS, not for Fedora as a whole.  Versions of FCOS based on Fedora 34 are no longer supported.

**Comment by ShamimIslam** [comment]
So will Fedora 35 work with Clevis/Tang?

**Comment by dustymabe** [comment]
> So will Fedora 35 work with Clevis/Tang?

If I understand correctly, the [latest released versions of FCOS](https://getfedora.org/en/coreos?stream=stable) should work with Clevis/Tang.

**Comment by bgilbert** [comment]
The discussion has strayed from the original bug, so I'm going to go ahead and lock it as off-topic.  Please seek any further support via the appropriate channels.
